parcels/particlefile/baseparticlefile.py

```diff
@@ -93,16 +93,22 @@ def __init__(self, name, particleset, outputdt=np.infty, chunks=None, write_onde
                                np.int64: np.iinfo(np.int64).max, np.uint8: np.iinfo(np.uint8).max,
                                np.uint16: np.iinfo(np.uint16).max, np.uint32: np.iinfo(np.uint32).max,
                                np.uint64: np.iinfo(np.uint64).max}
-
-        extension = os.path.splitext(str(name))[1]
-        if extension in ['.nc', '.nc4']:
-            raise RuntimeError('Output in NetCDF is not supported anymore. Use .zarr extension for ParticleFile name.')
-        if MPI and MPI.COMM_WORLD.Get_size() > 1:
-            self.fname = os.path.join(name, f"proc{self.mpi_rank:02d}.zarr")
-            if extension in ['.zarr']:
-                logger.warning(f'The ParticleFile name contains .zarr extension, but zarr files will be written per processor in MPI mode at {self.fname}')
+        if issubclass(type(name), zarr.storage.Store):
+            # If we already got a Zarr store, we won't need any of the naming logic below.
+            # But we need to handle incompatibility with MPI mode for now:
+            if MPI:
```

Comment:
Also need to check if we're actually in MPI mode (i.e. more than one processor)
```suggestion
            if MPI and MPI.COMM_WORLD.Get_size() > 1:
```